web/src/AnalyticsNudge.test.tsx

```diff
@@ -22,18 +22,30 @@ it("shows request-in-progress message", () => {
   expect(component).toMatchSnapshot()
 })
 
-it("shows success message", () => {
+it("shows success message: opt out", () => {
   const component = shallow(<AnalyticsNudge needsNudge={false} />)
 
-  component.setState({ requestMade: true, responseCode: 200 })
+  component.setState({ optIn: false, requestMade: true, responseCode: 200 })
 
   expect(component).toMatchSnapshot()
 })
 
-it("shows failure message", () => {
+it("shows success message: opt in", () => {
   const component = shallow(<AnalyticsNudge needsNudge={false} />)
 
-  component.setState({ requestMade: true, responseCode: 418 })
+  component.setState({ optIn: true, requestMade: true, responseCode: 200 })
+
+  expect(component).toMatchSnapshot()
+})
+
+it("shows failure message with response body", () => {
+  const component = shallow(<AnalyticsNudge needsNudge={false} />)
+
+  component.setState({
+    requestMade: true,
+    responseCode: 418,
```

Comment:
Is this [a reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418)? :P

Reply:
but of course!